### ISSUE TITLE
Unable to focus between multiple Emacs windows

### ISSUE THREAD

**Issue by akirakyle** [koekeishiya/chunkwm#283]
Hi,
I'm having an issue with Emacs 26.0.90 where I've set`(setq default-frame-alist '((undecorated . t)` to disable the title bar. As #105, I've set `chunkc tiling::rule --owner "Emacs.*" --state tile` in my `.chunkwmrc` and so `chunkwm` is successfully tiling Emacs windows. However the issue is that when there are multiple Emacs windows on a monitor, I am unable to focus between them using any of the `chunkc tiling::window --focus` commands (they work for all other windows). When I have windows from other applications and Emacs windows, the focus works between the non-Emacs windows and one of the Emacs windows but ignores the other Emacs windows. If I click on one of the windows that it is ignoring, now that window becomes the one Emacs window that `chunkwm` will focus. Furthermore with ffm, when I move my cursor over one of the non-focusing Emacs windows, the mouse will jump to the middle of the one focusing Emacs windows. The Emacs windows have the following properties
```
id: 5691
level: 0
name: .khdrc
owner: Emacs
role: AXTextField
subrole: AXDialog
movable: 1
resizable: 1
```
If I focus back and forth from the one focusable Emacs window (id 5691) I get the following output:
```
chunkwm-tiling command: (5) 'window --focus east'
3:desktop   5666:0:Terminal:akyle — chunkwm — ttys000 — 99×55 — ⌘1
3:desktop   5691:0:Emacs:.khdrc
3:desktop   5798:0:Emacs:.khdrc
39863:Emacs activated
35544:Terminal deactivated
chunkwm-tiling command: (5) 'window --focus west'
3:desktop   5691:0:Emacs:.khdrc
3:desktop   5666:0:Terminal:akyle — chunkwm — ttys000 — 99×55 — ⌘1
3:desktop   5798:0:Emacs:.khdrc
Terminal:akyle — chunkwm — ttys000 — 99×55 — ⌘1:5666 window focused
35544:Terminal activated
39863:Emacs deactivated
```
If I move the cursor back and forth from the one focusable Emacs window (id 5691):
```
39863:Emacs activated
35544:Terminal deactivated
35544:Terminal activated
Terminal:akyle — chunkwm — ttys000 — 99×55 — ⌘1:5666 window focused
39863:Emacs deactivated
```
So it seems that Emacs is only getting activated or deactivated which makes it seem like the window is being focused but chunkwm is unable to actually focus its windows.

When I click on the unfocusable window chunkwm outputs the following:
```
Emacs:.khdrc:5798 window focused
```

I can potentially make a screen recording if my explanation wasn't clear.

Thanks for making chunkwm! Its the only thing I've found that makes window management bearable on macOS

**Comment by koekeishiya** [comment]
I don't particularly know how Emacs works and I don't use it myself, although it appears to do various things that cause it to act weird in regards to how the windowing works. 

Only thing I can recommend is to try and run a clean, non-modified version of Emacs and see if that works as expected, and if it does, try to identify what customization you may have applied that causes this issue to exist.

I'm sure there are people that are successfully using Emacs with chunkwm.

**Comment by akirakyle** [comment]
Yeah the current state of Emacs' GUI code is a bit of a mess on every platform (including GNU/Linux) so its not surprising many people have trouble with it. The clean, unmodified version of emacs does tile and focus correctly when the appropriate rule is applied. However the development release (26.0.90) includes a new feature to disable the title bar which results in this focusing issue. #265 is a different bug that I didn't encounter but is also caused by the same feature in the 26.0.90 development Emacs.

I understand that this isn't a bug you have an incentive to deal with since you don't use Emacs, so feel free to close this issue. 

If I get a chance in the future to look more into this, I'll post back here with the outcome for those who do use Emacs. In any case it seems Emacs windows without a title bar should not have the subrole of `AXDialog`, which is something that would need to be fixed in Emacs' code.

**Comment by dominiklohmann** [comment]
Honestly at this point, why not just add a tiling rule for `owner=Emacs` and `subrole=AXDialog` to tile by default?

The Emacs developers do not seem to adhere to macOS programming guidelines for user interfaces, so I doubt there is much that can be done here besides that.

**Comment by akirakyle** [comment]
I do have a rule for Emacs and it does correctly tile even without a title bar when the subrole is `AXDialog`. The issue I'm encountering as I described above is that chunkwm seems to be unable to focus between multiple emacs windows on the same display. If you look at the output of chunkwm I copied above, thats because chunkwm only activates emacs, it doesn't actually focus any of its windows, which is weird since it still tiles them correctly.

**Comment by CestDiego** [comment]
I believe it's important to be able to have some Emacs frames (UI Windows for non-emacs users) float instead of tiling. I used this heavily when using org-capture with org-protocol to capturing notes when I was using Linux (http://www.diegoberrocal.com/blog/2015/08/19/org-protocol/) 

Also we want Emacs tooltips to not tile, I had a problem before in which the tiling rules were also affecting the tooltips, and it made chunkwm unusable until I changed that rule.

Where can we submit a bug on the Emacs for macOS build so that we can make sure they adhere somewhat to the macOS dev guidelines?

**Comment by kornosaurus** [comment]
I seem to have the same problem with undecorated Alacritty windows https://github.com/koekeishiya/chunkwm/issues/309
